test/bats/aws.bats

```diff
@@ -26,16 +31,138 @@ export PM_TEST_LONG_NAME=ParameterStoreTestWithLongName-$UUID
 export PM_ROTATION_TEST_NAME=ParameterStoreRotationTest-$UUID
 
 setup_file() {
-   #Create test secrets
-   aws secretsmanager create-secret --name $SM_TEST_1_NAME --secret-string SecretsManagerTest1Value --region $REGION
-   aws secretsmanager create-secret --name $SM_TEST_2_NAME --secret-string SecretsManagerTest2Value --region $REGION
-   aws secretsmanager create-secret --name $SM_SYNC_NAME --secret-string SecretUser --region $REGION
 
-   aws ssm put-parameter --name $PM_TEST_1_NAME --value ParameterStoreTest1Value --type SecureString --region $REGION
-   aws ssm put-parameter --name $PM_TEST_LONG_NAME --value ParameterStoreTest2Value --type SecureString --region $REGION
+  export AWS_USER_PAS_POLICY="${CLUSTER_NAME:0:12}-ParameterAndSecret-access-${UUID}"
+  export CSI_APP_ROLE_NAME="${CLUSTER_NAME:0:12}-csi-app-role-${UUID}"
+  export CSI_APP_POLICY_NAME="${CLUSTER_NAME:0:12}-csi-app-policy-${UUID}"
+
+  cat > $BATS_TEST_DIR/aws-user-pas-policy.json <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "AllowPutGetDeleteSpecificSSMParameters",
+      "Effect": "Allow",
+      "Action": [
+        "ssm:PutParameter",
+        "ssm:GetParameter",
+        "ssm:DeleteParameter"
+      ],
+      "Resource": [
+        "arn:aws:ssm:${REGION}:${ACCOUNT_NUMBER}:parameter/${PM_TEST_1_NAME}",
+        "arn:aws:ssm:${REGION}:${ACCOUNT_NUMBER}:parameter/${PM_TEST_LONG_NAME}",
+        "arn:aws:ssm:${REGION}:${ACCOUNT_NUMBER}:parameter/${PM_ROTATION_TEST_NAME}"
+      ]
+    },
+    {
+      "Sid": "AllowCreateGetDeleteSpecificSecrets",
+      "Effect": "Allow",
+      "Action": [
+        "secretsmanager:CreateSecret",
+        "secretsmanager:GetSecretValue",
+        "secretsmanager:DeleteSecret",
+        "secretsmanager:PutSecretValue"
+      ],
+      "Resource": [
+        "arn:aws:secretsmanager:${REGION}:${ACCOUNT_NUMBER}:secret:${SM_TEST_1_NAME}*",
+        "arn:aws:secretsmanager:${REGION}:${ACCOUNT_NUMBER}:secret:${SM_TEST_2_NAME}*",
+        "arn:aws:secretsmanager:${REGION}:${ACCOUNT_NUMBER}:secret:${SM_SYNC_NAME}*",
+        "arn:aws:secretsmanager:${REGION}:${ACCOUNT_NUMBER}:secret:${SM_ROT_TEST_NAME}*"
+      ]
+    }
+  ]
+}
+EOF
+
+  PAS_POLICY=$(aws iam create-policy --policy-name "${AWS_USER_PAS_POLICY}" \
+  --policy-document file://$BATS_TEST_DIR/aws-user-pas-policy.json \
+  --query 'Policy.Arn' --output text)
+
+  aws iam attach-user-policy --user-name $AWS_USER_NAME --policy-arn $PAS_POLICY
```

Comment:
The IAM user should have the required permissions to create, get, and delete secrets.